terraform/groups/ecs-service/variables.tf

```diff
@@ -0,0 +1,135 @@
+# ------------------------------------------------------------------------------
+# Environment
+# ------------------------------------------------------------------------------
+variable "environment" {
+  type        = string
+  description = "The environment name, defined in envrionment vars."
+}
+variable "aws_region" {
+  default     = "eu-west-2"
+  type        = string
+  description = "The AWS region for deployment."
+}
+variable "aws_profile" {
+  default     = "development-eu-west-2"
+  type        = string
+  description = "The AWS profile to use for deployment."
+}
+
+# ------------------------------------------------------------------------------
+# Docker Container
+# ------------------------------------------------------------------------------
+variable "docker_registry" {
+  type        = string
+  description = "The FQDN of the Docker registry."
+}
+
+# ------------------------------------------------------------------------------
+# Service performance and scaling configs
+# ------------------------------------------------------------------------------
+variable "desired_task_count" {
+  type        = number
+  description = "The desired ECS task count for this service"
+  default     = 1 # defaulted low for dev environments, override for production if required
+}
+variable "max_task_count" {
+  type        = number
+  description = "The maximum number of tasks for this service."
+  default     = 3
+}
+variable "required_cpus" {
+  type        = number
+  description = "The required cpu resource for this service. 1024 here is 1 vCPU"
+  default     = 256 # defaulted minimum required for fargate services, override for production
```

Comment:
This is not overridden for live - is this intentional or is it OK that it uses the default values?

Reply:
Or is this set elsewhere?

Reply:
I suppose this would be a load specific question and whether this service would struggle with this current CPU at live. Best to consult your architect to see whether you need to give this more resource.

Reply:
thanks @JoelClemence @Ngriffiths1 I have sent a message to our architects.

Reply:
As per billy's suggestions these values for CPU and memory need to be tested with the figures he suggested i.e 8cpu/16gb. 1024 here is 1 vCPU so you'll need 8* 1024. here is the documentation for AWS fargate values for more information https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-tasks-services.html#fargate-tasks-size

Reply:
thanks @Ngriffiths1 I have updated it.